components/eam/src/physics/crm/crm_physics.F90

```diff
@@ -201,45 +311,74 @@ subroutine crm_physics_register()
+   call pbuf_add_field('RPRDTOT',    'physpkg',dtype_r8,(/pcols,pver/),rprdtot_idx )
+   call pbuf_add_field('CLDTOP',     'physpkg' ,dtype_r8,(/pcols,1/),  cldtop_idx )
+   call pbuf_add_field('CLDBOT',     'physpkg' ,dtype_r8,(/pcols,1/),  cldbot_idx )
+   call pbuf_add_field('NEVAPR_SHCU','physpkg',dtype_r8,(/pcols,pver/),nevapr_shcu_idx )
+   call pbuf_add_field('PREC_SH',    'physpkg',dtype_r8,(/pcols/),     prec_sh_idx )
+   call pbuf_add_field('SNOW_SH',    'physpkg',dtype_r8,(/pcols/),     snow_sh_idx )
+
+   call pbuf_add_field('cush',       'global'  ,dtype_r8,(/pcols,dyn_time_lvls/), idx )  
+
+   call pbuf_add_field('AST',      'global',  dtype_r8, (/pcols,pver,dyn_time_lvls/), ast_idx)
+   call pbuf_add_field('FICE',     'physpkg', dtype_r8, (/pcols,pver/), fice_idx)
+
+   !----------------------------------------------------------------------------
+   !----------------------------------------------------------------------------
+
 end subroutine crm_physics_register
 
 !===================================================================================================
 !===================================================================================================
 
-subroutine crm_physics_init(state,species_class)
+subroutine crm_physics_init(state, pbuf2d, species_class)
 !---------------------------------------------------------------------------------------------------
 ! 
 ! Purpose: initialize some variables, and add necessary fields into output fields 
 !
 !---------------------------------------------------------------------------------------------------
-   use physics_buffer,        only: pbuf_get_index
+   use time_manager,          only: is_first_step
+   use physics_buffer,        only: physics_buffer_desc, pbuf_get_index, pbuf_set_field
    use phys_control,          only: phys_getopts
    use crm_history,           only: crm_history_init
+   use cam_history,    only: addfld
+   use constituents,   only: apcnst, bpcnst, cnst_name, cnst_longname, cnst_get_ind
 #ifdef ECPP
    use module_ecpp_ppdriver2, only: papampollu_init
 #endif
    use ppgrid,                only: begchunk, endchunk
    use constituents,          only: pcnst, cnst_get_ind
    !----------------------------------------------------------------------------
    ! interface variables
-   ! species_class is defined as input so it needs to be outside 
-   ! of MODAL_AERO condition for 1-moment micro to work
+   ! NOTE - species_class is an input so it needs to be outside of ifdef MODAL_AERO for 1-mom micro
    type(physics_state), intent(inout), dimension(begchunk:endchunk) :: state
-   integer, intent(inout) :: species_class(:)
+   type(physics_buffer_desc), pointer :: pbuf2d(:,:)
+   integer, intent(inout) :: species_class(:) 
    !----------------------------------------------------------------------------
    ! local variables
-   integer :: m
+   integer :: m, mm
+   integer :: ncnst = 8
```

Comment:
Can we grab this from above rather than hard-coding the value?

Reply:
I took this from micro_mg_cam.F90 where they hardcoded it for different MG options. The parameter is specific to the microphysics, so I don't think there's anywhere to grab the value from that would be appropriate. It has to match the cnst_names array which is hard coded at the top of the module, so it might be better to just grab the length of that array.

Reply:
Fixed in recent commit